src/control/cmd/dmg/server.go

```diff
@@ -0,0 +1,58 @@
+//
+// (C) Copyright 2021 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package main
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/lib/control"
+)
+
+// ServerCmd is the struct representing the top-level server subcommand.
+type ServerCmd struct {
+	SetLogMasks serverSetLogMasksCmd `command:"set-logmasks" description:"Set log masks for a set of facilities to a given level on all DAOS system engines at runtime"`
+}
+
+// serverSetLogMasksCmd is the struct representing the command to set engine log
+// levels at runtime across system.
+type serverSetLogMasksCmd struct {
+	logCmd
+	ctlInvokerCmd
+	hostListCmd
+	jsonOutputCmd
+	Masks string `long:"masks" alias:"m" description:"Set log masks for a set of facilities to a given level. The input string should look like: PREFIX1=LEVEL1,PREFIX2=LEVEL2,... where the syntax is identical to what is expected by 'D_LOG_MASK' environment variable. If unset then reset engine log masks to use the 'log_mask' value set in the server config file (for each engine) at the time of DAOS system format"`
```

Comment:
I don't see where there is any validation being done on the supplied masks. At this level, I don't think we should be checking for known prefixes, etc, but I do think we should at least validate that the string is well-formed (i.e. no crazy characters, not overly long, has the correct delimiters, etc). That validation should probably happen in the control API function.

Reply:
the validation is currently done server side, I don't necessarily see what we gain by doing client side validation, can I address that in a follow-up PR

Reply:
ticket: https://jira.hpdd.intel.com/browse/DAOS-8200

Reply:
It's just a defense-in-depth practice. We should ideally prohibit obviously-wrong inputs from getting across the RPC boundary and in particular down into the engine.